### PR TITLE
feat: search_user_groups接口功能扩展 --story=117025142

### PR DESCRIPTION
**Pull request by lthon-sha**

### CODE REVIEW

**Comment by lthon-sha** [comment]
Pasted image 20240417152808

**Comment by lthon-sha** [comment]
Pasted image 20240417154050

File: bkmonitor/kernel_api/views/v4/notice_group.py [hunk 1]
```diff
@@ -324,6 +325,9 @@ def perform_request(self, params):
         if params.get("ids"):
             user_groups = user_groups.filter(id__in=params.get("ids"))
 
+        if params.get("name"):
+            user_groups = user_groups.filter(name__icontains=params["name"])
```
Comment (lthon-sha) [inline bkmonitor/kernel_api/views/v4/notice_group.py:329]: 搜索不区分大小写

**Comment by lthon-sha** [comment]
image

**Review by chenguo367 — APPROVED** [review]

**Pull request merged**